Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1003 @@
+      ContinuousMap.smul_apply IsROrC.I]
+  · exact FunLike.congr_fun hf_re x
+  · rw [← IsROrC.I_im' (f x)]
+    congr! 1
+    exact FunLike.congr_fun hf_im x
+#align cfc₂_range cfc₂_range
+
+/-- For an isometric continuous functional calculus for `a` over `is_R_or_C 𝕜`, the range is
+precisely the `elemental_star_algebra` generated by `a`. -/
+theorem cfc₁_exists_of_mem_elementalStarAlgebra {𝕜 A : Type _} [IsROrC 𝕜] [NormedRing A]
+    [StarRing A] [NormedAlgebra 𝕜 A] [StarModule 𝕜 A] [NormedStarGroup A] {a : A}
+    [CompactSpace (spectrum 𝕜 a)] [CFCClass 𝕜 a] {x : A} (hx : x ∈ elementalStarAlgebra 𝕜 a) :
+    ∃ f : C(spectrum 𝕜 a, 𝕜), cfc₁ 𝕜 a f = x := by
+  rwa [← cfc₁_range] at hx
+#align cfc₁_exists_of_mem_elemental_star_algebra cfc₁_exists_of_mem_elementalStarAlgebra
+
+theorem cfc₂_exists_of_mem_elementalStarAlgebra {𝕜 A : Type _} [IsROrC 𝕜] [NormedRing A]
+    [StarRing A] [NormedAlgebra 𝕜 A] [StarModule 𝕜 A] [NormedStarGroup A] {a : A}
+    [CompactSpace (spectrum 𝕜 a)] [CFCClass 𝕜 a] {x : A} (hx : x ∈ elementalStarAlgebra 𝕜 a) :
+    ∃ f : C(𝕜, 𝕜), cfc₂ 𝕜 a f = x := by
+  rwa [← cfc₂_range] at hx
+#align cfc₂_exists_of_mem_elemental_star_algebra cfc₂_exists_of_mem_elementalStarAlgebra
+
+theorem cfc₁_map_spectrum (R : Type _) {A : Type _} [CommSemiring R] [StarRing R]
+    [TopologicalSpace R] [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A]
+    [TopologicalSpace A] [Algebra R A] (a : A) [CFCClass R a] (f : C(spectrum R a, R)) :
+    spectrum R (cfc₁ R a f) = Set.range f :=
+  CFCClass.hom_map_spectrum f
+#align cfc₁_map_spectrum cfc₁_map_spectrum
+
+theorem cfc₂_map_spectrum {R A : Type _} [CommSemiring R] [StarRing R] [TopologicalSpace R]
+    [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A] [TopologicalSpace A]
+    [Algebra R A] (a : A) [CFCClass R a] (f : C(R, R)) :
+    (spectrum R a).MapsTo f (spectrum R (cfc₂ R a f)) := by
+  rw [cfc₂, StarAlgHom.coe_comp, Function.comp_apply, cfc₁_map_spectrum]
+  exact fun x hx => ⟨⟨x, hx⟩, rfl⟩
+#align cfc₂_map_spectrum cfc₂_map_spectrum
+
+theorem cfc₂_map_spectrum' {R A : Type _} [CommSemiring R] [StarRing R] [TopologicalSpace R]
+    [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A] [TopologicalSpace A]
+    [Algebra R A] (a : A) [CFCClass R a] (f : C(R, R)) :
+    spectrum R (cfc₂ R a f) = f '' spectrum R a := by
+  rw [cfc₂, StarAlgHom.coe_comp, Function.comp_apply, cfc₁_map_spectrum]
+  ext
+  constructor
+  · rintro ⟨x, rfl⟩
+    exact ⟨x, x.prop, rfl⟩
+  · rintro ⟨x, hx, rfl⟩
+    exact ⟨⟨x, hx⟩, rfl⟩
+#align cfc₂_map_spectrum' cfc₂_map_spectrum'
+
+theorem cfc₂_comp {R A : Type _} [CommSemiring R] [StarRing R] [TopologicalSpace R]
+    [TopologicalSemiring R] [ContinuousStar R] [Ring A] [StarRing A] [TopologicalSpace A]
+    [Algebra R A] (a : A) [CFCClass R a] (f g : C(R, R)) [CFCCoreClass R (cfc₂ R a f)]
+    [Subsingleton (CFCCoreClass R (cfc₂ R a f))] : cfc₂ R a (g.comp f) = cfc₂ R (cfc₂ R a f) g := by
```

Comment:
Do you think it would be useful to have a version saying `cfc₁ R a (g.comp f) = cfc₂ R (cfc₁ R a f) g`?

Reply:
Try writing it down (the statement) and then you'll see the problem.

Reply:
Yes, I think that's definitely worthwhile. I'll add it.

Reply:
@ADedecker I actually ended up adding the version for `cfc₁` anyway, because it's essentially what was going on in the proofs of the others. So now there's a `cfc₁_comp`, a `cfc₁₂_comp` and a `cfc₂_comp`.